tests/vmi_configuration_test.go

```diff
@@ -2361,4 +2361,26 @@ var _ = Describe("Configurations", func() {
 			table.Entry("[test_id:3808] Should be rejected when using ide", "ide", "IDE bus is not supported"),
 		)
 	})
+
+	It("VMI with masquerade binding and guest agent should expose Pod IP as its public address", func() {
+		vmi := tests.NewRandomFedoraVMIWitGuestAgent()
+
+		By("Starting a VirtualMachineInstance")
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred(), "Should successfully create VMI")
+		tests.WaitForSuccessfulVMIStart(vmi)
+
+		vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred(), "Should successfully get VMI")
+
+		tests.WaitAgentConnected(virtClient, vmi)
+
+		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+		Eventually(func() string {
```

Comment:
Why do we we need Eventually here? What are we waiting for exactly?

Reply:
on L#2376 ensuring that the Agent is running helps, but there's a still finite amount of time between when the agent reports as ready and when the IP gets updated, so this is needed to reduce flaky results

Reply:
I still don't get the Eventually. The IP is reported there even before the guest agent has started since we use PodIP, right? So you don't need to wait at all to successfully pass the check.

Isn't Consistently what you are looking for? To make sure that the IP remains on PodIP even once the agent kicks in?

Reply:
@phoracek Yes, just added `Consistently` block also

Reply:
I still don't get why Eventually is needed here.

Before the guest agent kicks in, PodIP is reported as VMs IP (that's the default). After guest agent kicks in, it is exactly the same, so there is nothing we can wait for with Eventually. The block would return right away on the original report.

Is there any need to keep the Eventually?

Reply:
@petrkotas I'm not sure about `Before the guest agent kicks in, PodIP is reported as VMs IP`
is this something guaranteed? Because without `Eventually` we will insert a flaky test here

Reply:
Is it flaky without Eventually? Or do you just expect it to be?

It seems like we report the IP once the VM Pod is ready: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vmi.go#L369

Since we wait for successful start here on https://github.com/kubevirt/kubevirt/pull/3063/files#diff-d47bf036797965d742615e64adc3b777R2371, I would expect it to be ready. But I don't know the codebase by heart, please convince me otherwise.

Reply:
@phoracek It was guess
I think we can remove `Eventually` because of this https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vmi.go#L369

Reply:
@phoracek Done!